drivers/platform/stm32/stm32_i2c.c

```diff
@@ -43,6 +43,52 @@
 #include "no_os_i2c.h"
 #include "stm32_i2c.h"
 
+/**
+ * @brief i2c_table contains the pointers towards the I2C buses
+*/
+static void *i2c_table[4];
+
+/**
+ * @brief STM32 I2C bus initializer
+*/
+int32_t stm32_i2c_initialize_bus(uint32_t bus_number,
+				 struct no_os_i2c_init_param *param)
```

Comment:
Here should I initialize a different spibus_init_param? For users this part is not exposed and allocating extra variable for this just makes the runtime allocation slower

Reply:
this is hidden from the user so you can simply use no_os_i2c_init_param, you're right